### PR TITLE
Migrate Site Management Service to WordPressComRestApi

### PR DESCRIPTION
**Pull request by SergioEstevao**
Refs #5245 

To test:
 - Login in the app using WP.com
 - Try export content from a site
 - Try delete a site.

Needs review: @koke

### CODE REVIEW

File: WordPress/Classes/Networking/SiteManagementServiceRemote.swift [hunk 1]
```diff
@@ -51,11 +51,11 @@ public class SiteManagementServiceRemote : ServiceRemoteREST
     ///
     public func exportContent(siteID: NSNumber, success: (() -> Void)?, failure: (NSError -> Void)?) {
         let endpoint = "sites/\(siteID)/exports/start"
-        let path = self.pathForEndpoint(endpoint, withVersion: ServiceRemoteRESTApiVersion_1_1)
+        let path = self.pathForEndpoint(endpoint, withVersion: ._1_1)
```
Comment (koke) [inline WordPress/Classes/Networking/SiteManagementServiceRemote.swift]: That `._1_1` looks really ugly in Swift, but not sure what's the best alternative. The one I'm liking more is to add another value to the enum that doesn't match the prefix so the generated Swift value is `Version_1_1`:

```obj-c
typedef NS_ENUM(NSInteger, ServiceRemoteWordPressComRESTApiVersion)
{
    // Don't use this value
    ServiceRemoteWordPressComRESTApiUnknownVersion = 0,
    ServiceRemoteWordPressComRESTApiVersion_1_1 = 1001,
    ServiceRemoteWordPressComRESTApiVersion_1_2 = 1002,
};
```

https://gist.github.com/koke/0b6d994aaf85bd8239524258b842fea3

**Comment by koke** [comment]
Travis is failing, and `Script/build.sh` is also failing to build locally

**Comment by SergioEstevao** [comment]
@koke ready for another look.

File: WordPress/WordPressApi/WordPressComRestApi.swift [hunk 2]
```diff
@@ -42,7 +42,7 @@ public final class WordPressComRestApi: NSObject
         }
         sessionConfiguration.HTTPAdditionalHeaders = additionalHeaders
         let sessionManager = AFHTTPSessionManager(baseURL:baseURL, sessionConfiguration:sessionConfiguration)
-        sessionManager.responseSerializer = WordPressComRestAPIResponseSerializer()
+        sessionManager.responseSerializer = WordPressComRestAPIResponseSerializer()
```
Comment (koke) [inline WordPress/WordPressApi/WordPressComRestApi.swift]: Some trailing whitespace got in here

**Comment by koke** [comment]
There's some extra whitespace, but other than that it looks good :shipit:

**Pull request merged**